packages/flutter_tools/test/general.shard/runner/target_devices_test.dart

```diff
@@ -1486,6 +1486,44 @@ Please choose one (or "q" to quit): '''));
           }, overrides: <Type, Generator>{
             AnsiTerminal: () => terminal,
           });
+
+
+          testUsingContext('handle invalid options for device', () async {
+            deviceManager.iosDiscoverer.deviceList = <Device>[nonEphemeralDevice];
+
+            final TestTargetDevicesWithExtendedWirelessDeviceDiscovery targetDevices = TestTargetDevicesWithExtendedWirelessDeviceDiscovery(
+              deviceManager: deviceManager,
+              logger: logger,
+            );
+            targetDevices.waitForWirelessBeforeInput = true;
+
+            // Having the `0` first is an invalid choice for a device, the second
+            // item in the list, `1` is a valid option though which will return a
+            // valid device
+            targetDevices.deviceSelection.input = <String>['0', '1'];
```

Comment:
Could you add another input here that's out of range on the top end too (so 2 in this case)?

Reply:
Oh good catch!